### ISSUE TITLE
How can I get bounding boxes and segmentation labels?

### ISSUE THREAD

**Issue by SoonminHwang** [openai/universe-windows-envs#6]
Hi guys, 
Thanks for your effort. I really appreciate your work.

In the openai [blog](https://openai.com/blog/GTA-V-plus-Universe/),

> The environment also enables collecting massive amounts of labelled data: you can use the underlying GTA V engine to collect 2D or 3D bounding boxes and segmentation labels for cars, pedestrians, bicycles, animals, road surface, traffic signs, or any one of GTA V's other 7000+ objects. 

So I have tried to get these labelled data from GTA-V today, but I cannot find the way to get this labels.
Also I'm so confused, because of the codes in the [technical report](https://drive.google.com/file/d/0B0zrU_fRAzNcOU1Hd2x0RFZFdjg/view).
I think the C# codes in the tech report are not compatible with universe, right?

Please help me.
Thanks in advance.

**Comment by jamesdenney** [comment]
From briefly looking over the technical report they used ScriptHookVNet to access GTAV game engine where Universe uses ScriptHookV.  ScriptHookVNet is just a wrapper around ScriptHookV allowing you to write code in C# vice C++.  There should be a 1 to 1 mapping of function calls between the 2.

**Comment by crizCraig** [comment]
yep, i have been going through @arturf1's C# as of late and am able to get 3d corners of the cars, but don't have bounding boxes working quite yet. i think there is some issue with determining whether the entity is visible on the screen, where some (new?) scripthookv functions like `ENTITY::IS_ENTITY_VISIBLE` for determining if the entity is visible may help. segmentation labels can be obtained via renderdoc using [this method](https://download.visinf.tu-darmstadt.de/data/from_games/). code link is available on their site.

**Comment by monajalal** [comment]
@crizCraig I have finished the setup and installation of "Playing for Data" which post-processes the captures and provides stuff like the ones I have attached here, however the code for annotation and segmentation is not released yet, unfortunately. So yours doesn't have semantic segmentation? or how do you detect if a car is a car? I don't really care if you have bounding boxes or so as long as you can detect if a car or human is in the captured frame.
[GTAVLauncher_2017.01.05_18.33.36_frame9946__mesh.txt](https://github.com/openai/universe-windows-envs/files/703612/GTAVLauncher_2017.01.05_18.33.36_frame9946__mesh.txt)
[GTAVLauncher_2017.01.05_18.33.36_frame9946__shader.txt](https://github.com/openai/universe-windows-envs/files/703610/GTAVLauncher_2017.01.05_18.33.36_frame9946__shader.txt)
[GTAVLauncher_2017.01.05_18.33.36_frame9946__tex.txt](https://github.com/openai/universe-windows-envs/files/703611/GTAVLauncher_2017.01.05_18.33.36_frame9946__tex.txt)
![gtavlauncher_2017 01 05_18 33 36_frame9946__final](https://cloud.githubusercontent.com/assets/1892917/21919928/821f4ad4-d923-11e6-902a-79491ed5ff73.png)

**Comment by monajalal** [comment]
@jamesdenney while worked on finishing the setup for Playing For Data customized renderdoc, I realized you can't have both ScriptHooKV, ScriptHookV.net and RenderDoc going on at the same time as it crashes the game. 
However, for me, both works separately. 
I wonder how you guys got that error fix? 
@SoonminHwang here's two great videos showing how Script Hook V 
https://www.youtube.com/watch?v=jn0iR8ptBP0
and Script Hook V.Net work:
https://www.youtube.com/watch?v=pjbLHIQZzdc&t=108s (installation)
https://www.youtube.com/watch?v=Bsu2FZB5uSg&t=28s (demo)
the C# code for stuntplanesmoke https://www.gta5-mods.com/scripts/stunt-plane-smoke-net

**Comment by jamesdenney** [comment]
@monajalal I have run into the issue of not being able to run ScriptHookV and RenderDoc at the same time before and sadly there is no good solution to the issue.  The issue is with how ScriptHookV tries to hook into graphics pipeline.  We don't have access to this code so we can't modify it in any way.

**Comment by 4dead-in-car-crash** [comment]
I've posted this issue on ScriptHookV's forum: http://gtaforums.com/topic/788343-script-hook-v/?p=1069351840 

The original DeepDrive DirectX hooks (based on OBS) worked in tandem with ScriptHook which meant multiple levels of misdirection, but seemed to work fine. I wonder if RenderDoc can me modified to work as well.
